### PR TITLE
Allows multitile to work with arbitrary list lengths

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
The multitile component now works with any length of lists and creates all of the fillers relative to MACH_CENTER, rather than the center of the table.

## Why It's Good For The Game
This makes it easier and more intuitive to create big machines of various shapes while also dramatically reducing the size of the table needed to create shapes of high aspect ratios.

## Images of changes
image

## Testing
I tested the following shapes by changing the filler's sprite to that of a wooden table:
Hourglass:

		list(1, 1, 1, 1, 1)

		list(	1, 1, 1)

		list(  MACH_CENTER)

		list(	1, 1, 1)

		list(1, 1, 1, 1, 1)

Hollow Diamond:

		list(		 	1)

		list(	 	1,	0, 	1)

		list(	   1, 0,	0, 	0, 1)

		list(   1, 0,  0,	0, 	0, 0, 1)

		list(        1, 0,	0, 	0, 1)

		list( 		1, 	0, 	1, )

		list(		   MACH_CENTER)

Snake:

		list(	          1, 0, 0, 0, 1, 0, 0),

		list(MACH_CENTER, 0, 1, 0, 1, 0, 1, 0, 1),

		list(	          0, 0, 1, 0, 0, 0, 1),

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
tweak: Changed the multitile component so that it could work with arbitrary shapes
/:cl:

### CODE REVIEW

**Comment by Migratingcocofruit** [comment]
As per DgamerL's request it will be limited to work with rectangles so formatting is more consistent and readable. 
Other functions are retained

File: code/datums/components/multitile.dm [hunk 1]
```diff
@@ -16,45 +16,64 @@
...
-	var/current_width = 0
+	var/current_width = 1
+	var/tile_index = 1
 
-	for(var/turf/filler_turf as anything in RANGE_TURFS(distance_from_center, owner))
-		if(new_filler_map[max_height - current_height][max_width - current_width]) // Because the `block()` proc always works from the bottom left to the top right, we have to loop through our nested lists in reverse
+	for(var/turf/filler_turf as anything in block( \
+	owner.x - offset_x - distance_from_center_x, owner.y + offset_y - distance_from_center_y, owner.z, \
+	owner.x - offset_x + distance_from_center_x, owner.y + offset_y + distance_from_center_y, owner.z, \
+	))
```
Comment (BiancaWilkson) [inline code/datums/components/multitile.dm:65]:
```suggestion
	for(var/turf/filler_turf as anything in block( 
	owner.x - offset_x - distance_from_center_x, owner.y + offset_y - distance_from_center_y, owner.z, 
	owner.x - offset_x + distance_from_center_x, owner.y + offset_y + distance_from_center_y, owner.z, 
	))
```

File: code/datums/components/multitile.dm [hunk 2]
```diff
@@ -16,45 +16,64 @@
...
+
+	for(var/i in 1 to length(new_filler_map))
+		if(length(new_filler_map[i] != max_width))
+			stack_trace("A multitile component was passed a list wich did not have the same length every row. Atom parent is: [parent]")
+			var/obj/machinery/machine = parent
+			machine.deconstruct()
+			return COMPONENT_INCOMPATIBLE
+
+		for(var/j in 1 to length(new_filler_map[i]))
+			if(new_filler_map[i][j] == 2)
```
Comment (BiancaWilkson) [inline code/datums/components/multitile.dm:41]:
```suggestion
			if(new_filler_map[i][j] == MACH_CENTER)
```
Am I understanding this code block correctly? If so this is better for readability

**Review by BiancaWilkson — CHANGES_REQUESTED** [review]
Just some minor readability nitpicks, everything looks and runs well!

File: code/datums/components/multitile.dm (hunk 1 above)
Comment (DGamerL) [reply code/datums/components/multitile.dm:65]: Does `block()` work nicely with multiline arguments?

**Review by DGamerL — COMMENTED** [review]
Yeah just needs wilk's stuff

File: code/datums/components/multitile.dm (hunk 1 above)
Comment (Migratingcocofruit) [reply code/datums/components/multitile.dm]: Tested, and it seems to work fine

File: code/datums/components/multitile.dm (hunk 2 above)
Comment (Migratingcocofruit) [reply code/datums/components/multitile.dm]: done

**Review by BiancaWilkson — APPROVED** [review]
Looks good!

**Review by DGamerL — APPROVED** [review]
I'm amazed `block()` handles that well

**Pull request merged**